src/internal/pjs/api_server.go

```diff
@@ -42,7 +46,15 @@ func (a *apiServer) CreateJob(ctx context.Context, request *pjs.CreateJobRequest
 		}
 		inputs = append(inputs, fileset.PinnedFileset(*inputID))
 	}
+	var parent pjsdb.JobID
+	if request.Context != "" {
+		parent, err = a.resolveJobCtx(ctx, request.Context)
+		if err != nil {
+			return nil, errors.Wrap(err, "walk job")
```

Comment:
wrap the error with information that the caller of this function doesn't know, namely the name of the function that we're calling without the caller of CreateJob's knowledge; "resolveJobCtx".